src/silx/io/url.py

```diff
@@ -367,7 +376,10 @@ def path(self) -> str:
                 else:
                     path = self.__scheme + ":///" + path
             else:
-                path = self.__scheme + ":" + path
+                if self.__scheme.startswith("http"):
+                    path = self.__scheme + "://" + path
+                else:
+                    path = self.__scheme + ":" + path
```

Comment:
Unless I missed something, this should be fine:

```suggestion
                path = self.__scheme + "://" + path
```